app/lib/actions/katello/applicability/repository/regenerate.rb

```diff
@@ -11,9 +11,18 @@ class Regenerate < Actions::EntryAction
           end
 
           def run
-            ::Katello::Repository.find(input[:repo_id]).content_facets.each do |facet|
-              ::Katello::EventQueue.push_event(::Katello::Events::GenerateHostApplicability::EVENT_TYPE, facet.host.id)
+            content_facets = []
+            repo = ::Katello::Repository.find(input[:repo_id])
+            content_facets << repo.content_facets
+            content_facets << repo.clones.collect { |clone| clone.content_facets }
```

Comment:
It would be a lot more efficient to look up all the clones and then query the content facets for all of them (so its not a host query per repo).  I looked and there's actually already a method for that:
Repository#hosts_with_applicability